src/Access/IAccessStorage.h

```diff
@@ -128,7 +132,9 @@ class IAccessStorage : public boost::noncopyable
     /// Throws an exception if the specified name already exists.
     UUID insert(const AccessEntityPtr & entity);
     std::optional<UUID> insert(const AccessEntityPtr & entity, bool replace_if_exists, bool throw_if_exists);
+    std::optional<UUID> insert(const AccessEntityPtr & entity, bool replace_if_exists, bool throw_if_exists, std::optional<UUID> set_id);
```

Comment:
The function gets optional id and returns optional id. It doesn't look very good, maybe it's better to change it into something like:

```
bool insert(const AccessEntityPtr & entity, std::optional<UUID> & id, bool replace_if_exists, bool throw_if_exists);
```

Reply:
I decided to add `const UUID & id` to the `insertImpl` function, so now it should be much more straightforward